ml-proto/src/check.ml

```diff
@@ -241,6 +239,13 @@ and check_arm c t ts arm =
   check_literal c [t] l;
   check_expr c (if fallthru then [] else ts) e
 
+and check_memop memop at =
+  require (memop.align = 1 || memop.align = 2 ||
+           memop.align = 4 || memop.align = 8)
```

Comment:
Since we're checking that the alignment is at most the access size below, would it make sense to just check that the alignment is a power of 2 here, instead of having an explicit list? That way it'll automatically generalize to SIMD types.